_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -1041,12 +1044,19 @@ VAAPIVideoCORE::DoFastCopyExtended(
     mfxFrameSurface1* pSrc)
 {
     mfxStatus sts;
+    mfxU8* srcPtr;
+    mfxU8* dstPtr;
+
+    sts = GetFramePointerChecked(pSrc->Info, pSrc->Data, &srcPtr);
+    MFX_CHECK(MFX_SUCCEEDED(sts), MFX_ERR_UNDEFINED_BEHAVIOR);
+    sts = GetFramePointerChecked(pDst->Info, pDst->Data, &dstPtr);
+    MFX_CHECK(MFX_SUCCEEDED(sts), MFX_ERR_UNDEFINED_BEHAVIOR);
 
     // check that only memId or pointer are passed
     // otherwise don't know which type of memory copying is requested
     if (
-        (!LumaIsNull(pDst) && NULL != pDst->Data.MemId) ||
-        (!LumaIsNull(pSrc) && NULL != pSrc->Data.MemId)
+        (nullptr != dstPtr && nullptr != pDst->Data.MemId) ||
```

Comment:
"nullptr != pDst->Data.MemId" Looks strange. Does everyone has to now that MemId is defined as HANDLE which is defined as void*? Better compare to 0 to avoid confusing readers, or introduce INVALID_MEMID.

Reply:
or maybe better to drop the actual value and use implicit conversion to ```bool```?